core/domain/skill_domain.py

```diff
@@ -304,6 +304,55 @@ def validate(self):
                     'received %s' % example)
             example.validate()
 
+        # TODO(tjiang11): Extract content ids to audio translations out into
+        # its own object to reuse throughout audio-capable structures.
+        if not isinstance(self.content_ids_to_audio_translations, dict):
+            raise utils.ValidationError(
+                'Expected state content_ids_to_audio_translations to be a dict,'
+                'received %s' % self.param_changes)
+        for (content_id, audio_translations) in (
+                self.content_ids_to_audio_translations.iteritems()):
+
+            if not isinstance(content_id, basestring):
+                raise utils.ValidationError(
+                    'Expected content_id to be a string, received: %s' %
+                    content_id)
+            if not isinstance(audio_translations, dict):
+                raise utils.ValidationError(
+                    'Expected audio_translations to be a dict, received %s'
+                    % audio_translations)
+
+            allowed_audio_language_codes = [
+                language['id'] for language in (
+                    constants.SUPPORTED_AUDIO_LANGUAGES)]
+            for language_code, translation in audio_translations.iteritems():
+                if not isinstance(language_code, basestring):
+                    raise utils.ValidationError(
+                        'Expected language code to be a string, received: %s' %
+                        language_code)
+
+                if language_code not in allowed_audio_language_codes:
+                    raise utils.ValidationError(
+                        'Unrecognized language code: %s' % language_code)
+
+                translation.validate()
+
+        explanation_content_id = set([self.explanation.content_id])
+        worked_example_content_ids = set([example.content_id for example
+                                          in self.worked_examples])
+        content_ids = explanation_content_id | worked_example_content_ids
```

Comment:
do we check somewhere that explanation_content_id and worked_example_content_ids doest have any common ids? (asking this because I'm not sure how we generate ids for worked_examples)

If not then maybe we can have a check for `explanation_content_id` is not a subset of `worked_example_content_ids`.